arbnode/transaction_streamer.go

```diff
@@ -1834,21 +1843,78 @@ func (s *TransactionStreamer) Start(ctxIn context.Context) error {
 }
 
 const ESPRESSO_TRANSACTION_SIZE_LIMIT int = 10 * 1024
+const MAX_ATTESTATION_QUOTE_SIZE int = 4 * 1024
 
-func buildHotShotPayload(msgs *[]arbostypes.L1IncomingMessage) (espressoTypes.Bytes, int) {
+func (t *TransactionStreamer) buildHotShotPayload(msgs *[]arbostypes.MessageWithMetadata, submittedTxnPos []arbutil.MessageIndex) (espressoTypes.Bytes, int) {
```

Comment:
My other questions/comments below are mainly to understand if the attestation quote over the payload can qualify as the unique identifier for the payload, or if we should also include the message pos in the payload as a sort of salt for the attestation quote.

Reply:
The attestation quote is not acting as the unique identifier. We need each transaction to be only included only once as a reason for each transaction we have included the `position`.